server.go

```diff
@@ -172,48 +176,72 @@ func (s *Server) handleConn(c *Conn) error {
 	}
 }
 
-// ListenAndServe listens on the network address s.Addr and then calls Serve
-// to handle requests on incoming connections.
-//
-// If s.Addr is blank and LMTP is disabled, ":smtp" is used.
-func (s *Server) ListenAndServe() error {
-	network := "tcp"
-	if s.LMTP {
+// Listen configures a listener according to the server configuration, without TLS
+// This listener can then be passed to Serve. If access to the listener is not needed, then the
+// combined function ListenAndServe can be used
+func (s *Server) Listen() (net.Listener, error) {
```

Comment:
Let's not expose this

Reply:
The other option would be to put the tests that test the listener setup inside the smtp package. Would that be okay?